keep/contextmanager/contextmanager.py

```diff
@@ -187,4 +195,38 @@ def load_step_context(self, step_id, step_results, step_conditions):
 
     # TODO - add step per alert?
     def get_step_context(self, step_id):
-        return {"step_id": step_id, "step_context": self._steps_context.get(step_id)}
+        return {"step_id": step_id, "step_context": self.steps_context.get(step_id)}
+
+    def __load_state(self):
+        if self.state_file:
+            # TODO - SQLite
+            try:
+                with open(self.state_file, "r") as f:
+                    self.state = json.load(f)
+            except:
+                self.state = {}
+
+    def get_last_alert_run(self, alert_id):
+        if alert_id in self.state:
+            return self.state[alert_id][-1]
+        # no previous runs
+        else:
+            return {}
+
+    def set_last_alert_run(self, alert_id, alert_context, alert_status):
```

Comment:
add docs + how does `set_last_alert_run` is being used from outer class? shouldn't it be private?

Reply:
it's being called from the alert manager, it's something you've defined, let's think about it?